nifi-nar-bundles/nifi-riemann-bundle/nifi-riemann-processors/src/main/java/org/apache/nifi/processors/riemann/PutRiemann.java

```diff
@@ -0,0 +1,376 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.riemann;
+
+import com.aphyr.riemann.Proto;
+import com.aphyr.riemann.Proto.Event;
+import com.aphyr.riemann.client.RiemannClient;
+import org.apache.nifi.annotation.behavior.DynamicProperty;
+import org.apache.nifi.annotation.behavior.SupportsBatching;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.annotation.lifecycle.OnStopped;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.PropertyValue;
+import org.apache.nifi.components.Validator;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+@Tags({"riemann", "monitoring", "metrics"})
+@DynamicProperty(name = "Custom Event Attribute", supportsExpressionLanguage = true,
+  description = "These values will be attached to the Riemann event as a custom attribute",
+  value = "Any value or expression")
+@CapabilityDescription("Send events to Riemann")
+@SupportsBatching
+public class PutRiemann extends AbstractProcessor {
+  protected enum Transport {
+    TCP, UDP
+  }
+
+  protected RiemannClient riemannClient = null;
+  protected Transport transport;
+
+  public static final Relationship REL_SUCCESS = new Relationship.Builder()
+    .name("success")
+    .description("Metrics successfully written to Riemann")
+    .build();
+
+  public static final Relationship REL_FAILURE = new Relationship.Builder()
+    .name("failure")
+    .description("Metrics which failed to write to Riemann")
+    .build();
+
+
+  public static final PropertyDescriptor RIEMANN_HOST = new PropertyDescriptor.Builder()
+    .name("Riemann Address")
+    .description("Hostname of Riemann server")
+    .required(true)
+    .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+    .build();
+
+  public static final PropertyDescriptor RIEMANN_PORT = new PropertyDescriptor.Builder()
+    .name("Riemann Port")
+    .description("Port that Riemann is listening on")
+    .required(true)
+    .defaultValue("5555")
+    .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
```

Comment:
Non-empty validator is redundant with the integer validator provided below and the fact that this value is also required